src/Sylius/Bundle/CoreBundle/PaymentRequest/Checker/PaymentRequestIntegrityChecker.php

```diff
@@ -29,16 +30,19 @@ public function __construct(
     public function check(PaymentRequestHashAwareInterface $command): PaymentRequestInterface
     {
         $hash = $command->getHash();
-        Assert::notNull($hash, 'Payment request hash cannot be null.');
 
         $paymentRequest = $this->paymentRequestRepository->findOneByHash($hash);
```

Comment:
To my understanding of how checkers should look, checker should receive loaded `$paymentRequest` and not handle not found exceptions.